### PR TITLE
UOE-10791: Ignored rateLimit field for aidem bidder

### PR DESCRIPTION
**Pull request by pm-priyanka-bagade**

### CODE REVIEW

File: modules/pubmatic/openwrap/adapters/default_bidder.go [hunk 1]
```diff
@@ -37,6 +37,9 @@ func prepareBidParamJSONDefault(params BidderParameters) (json.RawMessage, error
 			if err != nil && mapping.Required {
 				return nil, err
 			}
+			if params.AdapterName == "aidem" && mapping.KeyName == "rateLimit" {
```
[the remark this answers is not in the document]
Comment (pm-priyanka-bagade) [reply modules/pubmatic/openwrap/adapters/default_bidder.go]: Done

File: modules/pubmatic/openwrap/adapters/bidders.go [hunk 2]
```diff
@@ -704,3 +704,42 @@ func builderPGAMSSP(params BidderParameters) (json.RawMessage, error) {
...
+	//  len=0 (no mandatory params present)
+	if jsonStr.Len() == 1 {
+		return nil, fmt.Errorf(errMandatoryParameterMissingFormat, params.AdapterName, "'publisherId' and 'siteId'")
+	}
+
+	trimComma(&jsonStr)
+	jsonStr.WriteByte('}')
+
+	return jsonStr.Bytes(), nil
+}
```
Comment (pm-saurabh-narkhede) [inline modules/pubmatic/openwrap/adapters/bidders.go:745]: can we simplify this func like,
```func builderAidem(params BidderParameters) (json.RawMessage, error) {
	jsonStr := bytes.Buffer{}
	jsonStr.WriteByte('{')

	fields := []string{"publisherId", "siteId", "placementId"}
	mandatoryFields := map[string]bool{"publisherId": true, "siteId": true}

	for _, field := range fields {
		if value, ok := getString(params.FieldMap[field]); ok {
			fmt.Fprintf(&jsonStr, `"%s":"%s",`, field, value)
		} else if mandatoryFields[field] {
			return nil, fmt.Errorf(errMandatoryParameterMissingFormat, params.AdapterName, "'%s'", field)
		}
	}

	if jsonStr.Len() == 1 { // len=1 means only the '{' is present
		return nil, fmt.Errorf(errMandatoryParameterMissingFormat, params.AdapterName, "'publisherId' and 'siteId'")
	}

	trimComma(&jsonStr)
	jsonStr.WriteByte('}')

	return jsonStr.Bytes(), nil
}```

File: modules/pubmatic/openwrap/adapters/bidders_test.go [hunk 3]
```diff
@@ -3060,3 +3060,56 @@ func Test_builderPGAMSSP(t *testing.T) {
...
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := builderAidem(tt.args.params)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("builderAidem() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			AssertJSON(t, tt.want, got)
+		})
+	}
+}
```
Comment (pm-saurabh-narkhede) [inline modules/pubmatic/openwrap/adapters/bidders_test.go:3115]: Check `s2s_bidders.json`, Can we add test case in this?

File: modules/pubmatic/openwrap/adapters/bidders.go [hunk 4]
```diff
@@ -704,3 +704,42 @@ func builderPGAMSSP(params BidderParameters) (json.RawMessage, error) {
 	jsonStr.WriteByte('}')
 	return jsonStr.Bytes(), nil
 }
+
+func builderAidem(params BidderParameters) (json.RawMessage, error) {
+	jsonStr := bytes.Buffer{}
+	jsonStr.WriteByte('{')
+
+	//publisherID
+	if publisherID, ok := getString(params.FieldMap["publisherId"]); ok {
```
Comment (AvinashKapre) [inline modules/pubmatic/openwrap/adapters/bidders.go:713]: use constant for ```publisherId```

File: modules/pubmatic/openwrap/adapters/bidders.go [hunk 5]
```diff
@@ -704,3 +704,42 @@ func builderPGAMSSP(params BidderParameters) (json.RawMessage, error) {
...
+		fmt.Fprintf(&jsonStr, `"publisherId":"%s",`, publisherID)
+	} else {
+		return nil, fmt.Errorf(errMandatoryParameterMissingFormat, params.AdapterName, "'publisherId'")
+	}
+
+	//siteId
+	if siteId, ok := getString(params.FieldMap["siteId"]); ok {
```
Comment (AvinashKapre) [inline modules/pubmatic/openwrap/adapters/bidders.go:720]: use constant for ```siteId```

Comment (pm-priyanka-bagade) [reply modules/pubmatic/openwrap/adapters/bidders.go]: modified

File: modules/pubmatic/openwrap/adapters/bidders.go (hunk 4 above)
Comment (pm-priyanka-bagade) [reply modules/pubmatic/openwrap/adapters/bidders.go]: modified

File: modules/pubmatic/openwrap/adapters/bidders.go (hunk 2 above)
Comment (pm-priyanka-bagade) [reply modules/pubmatic/openwrap/adapters/bidders.go:732]: modified

**Review by pm-saurabh-narkhede — APPROVED** [review]

**Review by AvinashKapre — APPROVED** [review]

**Review by pm-saurabh-narkhede — APPROVED** [review]

**Pull request merged**